### PR TITLE
removed OData from search result endpoint

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

File: Api/Controllers/AgentControllers/AvailabilitiesController.cs [hunk 1]
```diff
@@ -80,17 +80,16 @@ public async Task<IActionResult> GetAvailabilitySearchState([FromRoute] Guid sea
...
+        /// <param name="options">Pagination and filters</param>
         /// <returns>Availability results</returns>
         [HttpGet("searches/{searchId}")]
         [ProducesResponseType(typeof(IEnumerable<WideAvailabilityResult>), (int) HttpStatusCode.OK)]
         [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.BadRequest)]
         [MinCounterpartyState(CounterpartyStates.ReadOnly)]
         [InAgencyPermissions(InAgencyPermissions.AccommodationAvailabilitySearch)]
-        [EnableQuery(MaxAnyAllExpressionDepth = 2, EnsureStableOrdering = false)]
-        public async Task<IEnumerable<WideAvailabilityResult>> GetAvailabilitySearchResult([FromRoute] Guid searchId)
+        public async Task<IActionResult> GetAvailabilitySearchResult([FromRoute] Guid searchId, [FromQuery] SearchResultQueryOptions options)
```
Comment (mingazhev) [inline Api/Controllers/AgentControllers/AvailabilitiesController.cs:90]: `AvailabilitySearchFilter` will sound better I think

Comment (DFSko) [reply Api/Controllers/AgentControllers/AvailabilitiesController.cs]: Fixed

File: Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchService.cs [hunk 2]
```diff
@@ -120,10 +125,28 @@ IEnumerable<WideAvailabilityResult> CombineAvailabilities(IEnumerable<(Suppliers
...
                             searchSettings.IsSupplierVisible
                                 ? supplier
-                                : (Suppliers?) null,
+                                : (Suppliers?)null,
                             availability.HtId);
                     })
-                    .Where(a => a.RoomContractSets.Any());
+                    .AsQueryable();
+
+                if (options.MinPrice.HasValue)
```
Comment (mingazhev) [inline Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchService.cs:133]: It's better to move this to the local function like "ApplyFilters"

**Review by mingazhev — APPROVED** [review]
With a small comment

File: Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchService.cs (hunk 2 above)
Comment (DFSko) [reply Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchService.cs:133]: Implement this in another branch

**Pull request merged**